src/AST/ASTExpressionStatement.ts

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) ZALORA
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+import { IToken as Token } from '../tokens';
+import ASTExpression from './ASTExpression';
+import ASTStatement from './ASTStatement';
+
+/**
+ * ASTExpressionStatement is the base node for expression statements.
+ * The "Select Expression" (e.g. `.<identifier>`) is an expression statement.
+ */
+class ASTExpressionStatement implements ASTStatement {
+  token: Token;
+  expression: ASTExpression | null;
```

Comment:
if we declare a variable to be optional in TypeScript which is like this `expression?: ASTExpression`, It would only support the variable to hold a value or undefined but not null. If we need the variable to support null, then we have to use union to support null.
[link here](https://stackoverflow.com/a/57243788)

Reply:
Why do we need to support `null` specifically?

Reply:
JS works like that, the variable will be undefined until the variable is not initialised.

Reply:
i was wondering why did we specifically needed `null` just where I got confused with `nil` on the reference spec should be same as the `null` in JS/TS. Now I got the point.